src/cmd/services/m3coordinator/downsample/metrics_appender.go

```diff
@@ -394,7 +394,8 @@ func (a *metricsAppender) SamplesAppender(opts SampleAppenderOptions) (SamplesAp
 	// Apply drop policies results
 	a.curr.Pipelines, dropApplyResult = a.curr.Pipelines.ApplyOrRemoveDropPolicies()
 
-	if len(a.curr.Pipelines) > 0 && !a.curr.IsDropPolicyApplied() {
+	// Skip sending to downsampler if there's a drop policy or no pipeline defined.
+	if len(a.curr.Pipelines) > 0 && !a.curr.IsDropPolicyApplied() && !a.curr.IsDefault() {
```

Comment:
shouldn't `len(a.curr.Pipelines) > 0` already prevent sending the default staged metadata?

Reply:
the default confusingly contains one PipelineMetadata (so len == 1) which in itself is "empty" / default.